runtests.py

```diff
@@ -290,8 +290,22 @@ def main(argv):
     if args.build_only:
         sys.exit(0)
     else:
-        __import__(PROJECT_MODULE)
-        test = sys.modules[PROJECT_MODULE].test
+        try:
+            test, version, mod_path = import_module()
+        except ImportError:
+            # this may fail when running with --no-build, so try to detect
+            # an installed scipy in a subdir inside a repo
+            dst_dir = os.path.join(ROOT_DIR, 'build', 'testenv')
+            from sysconfig import get_path
+            py_path = get_path('platlib')
+            site_dir = os.path.join(dst_dir, get_path_suffix(py_path, 3))
+            print("Trying to import scipy from development installed path at:",
+                  site_dir)
+            sys.path.insert(0, site_dir)
+            os.environ['PYTHONPATH'] = \
+                os.pathsep.join((site_dir, os.environ.get('PYTHONPATH', '')))
+            test, version, mod_path = import_module()
```

Comment:
I have modified `sys.path` here. I think that should find scipy. It's running fine on my machine:
```
$ python runtests.py -n -s io
Trying to import scipy from development installed path at: /home/admin-pc/Smitlunagariya/scipy/build/testenv/lib/python3.10/site-packages
Running tests for scipy version:1.9.0.dev0+1203.eb2ad0a, installed at:/home/admin-pc/Smitlunagariya/scipy/build/testenv/lib/python3.10/site-packages/scipy
=========================================== test session starts ============================================
platform linux -- Python 3.10.0, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/admin-pc/Smitlunagariya/scipy, configfile: pytest.ini
plugins: cov-3.0.0, forked-1.4.0, xdist-2.5.0
collected 607 items                              
```

Reply:
I found the problem - it was mixed with another build issue which caused a problem. With that fix applied, this works for me.